x-pack/plugins/security_solution/public/common/components/exceptions/viewer/index.tsx

```diff
@@ -4,96 +4,411 @@
+                  defaultMessage="rule settings"
+                />
+              </EuiLink>
+            ),
+          }}
+        />
+      );
+    } else {
+      return <></>;
+    }
+  }, [filterOptions.showEndpointList, filterOptions.showDetectionsList, ruleSettingsUrl]);
+
+  const showEmpty = useMemo((): boolean => {
+    return !initLoading && !loadingList && exceptions.length === 0;
+  }, [initLoading, exceptions.length, loadingList]);
 
   return (
-    <EuiPanel paddingSize="none">
-      <EuiFlexGroup direction="column" gutterSize="none">
-        <EuiFlexItem>
-          <EuiFlexGroup direction="row">
-            <ExceptionDetails
-              showComments={showComments}
-              exceptionItem={exceptionItem}
-              onCommentsClick={onCommentsClick}
+    <>
+      {isModalOpen && (
+        <EuiOverlayMask>
+          <EuiModal onClose={onCloseExceptionModal}>
+            <EuiModalBody>
+              <EuiCodeBlock language="json" fontSize="m" paddingSize="m" overflowHeight={300}>
+                {`Modal goes here`}
+              </EuiCodeBlock>
+            </EuiModalBody>
+          </EuiModal>
+        </EuiOverlayMask>
+      )}
+
+      <Panel loading={initLoading}>
+        {initLoading && <Loader data-test-subj="loadingPanelAllRulesTable" overlay size="xl" />}
+
+        <ExceptionsViewerHeader
+          isInitLoading={initLoading}
+          supportedListTypes={availableListTypes}
+          detectionsListItems={detectionsList != null ? detectionsList.totalItems : 0}
+          endpointListItems={endpointList != null ? endpointList.totalItems : 0}
+          onFilterChange={onFiltersChange}
+          onAddExceptionClick={onAddException}
+        />
+
+        {(filterOptions.showEndpointList || filterOptions.showDetectionsList) && (
+          <>
+            <EuiSpacer size="xs" />
+            <StyledText size="s">{exceptionsSubtext}</StyledText>
+          </>
+        )}
+
+        <EuiSpacer size="l" />
+
+        <UtilityBar>
+          <UtilityBarSection>
+            <UtilityBarGroup>
+              <UtilityBarText dataTestSubj="showingRules">
+                {i18n.SHOWING_EXCEPTIONS(pagination.totalItemCount ?? 0)}
+              </UtilityBarText>
+            </UtilityBarGroup>
+
+            <UtilityBarGroup>
+              <UtilityBarAction iconSide="left" iconType="refresh" onClick={onFetchList}>
+                {i18n.REFRESH}
+              </UtilityBarAction>
+            </UtilityBarGroup>
+          </UtilityBarSection>
+        </UtilityBar>
+
+        <EuiSpacer size="s" />
+
+        <MyExceptionsContainer className="eui-yScrollWithShadows">
+          {showEmpty && (
+            <EuiEmptyPrompt
+              iconType="advancedSettingsApp"
+              title={<h2>{i18n.EXCEPTION_EMPTY_PROMPT_TITLE}</h2>}
+              body={<p>{i18n.EXCEPTION_EMPTY_PROMPT_BODY}</p>}
+              data-test-subj="exceptionsEmptyPrompt"
             />
```

Comment:
Since the parent container has a max-height I think we'd be okay to vertically center the empty prompt.

Reply:
Good call. Added to follow up PR.